sw/device/lib/runtime/otbn.h

```diff
@@ -32,13 +32,30 @@ typedef struct otbn_app {
    */
   const uint8_t *imem_end;
   /**
-   * Start of OTBN data memory.
+   * Start of initialized OTBN data memory.
+   *
+   * Data in this section is copied into DMEM when the app is loaded.
+   */
+  const uint8_t *dmem_data_start;
+  /**
+   * End of initialized OTBN data memory.
+   */
+  const uint8_t *dmem_data_end;
+  /**
+   * Start of uninitialized OTBN data memory.
+   *
+   * Data in this section is initialized to zeroes in DMEM when the app is
+   * loaded.
+   */
+  const uint8_t *dmem_bss_start;
+  /**
+   * End of uninitialized OTBN data memory.
    */
-  const uint8_t *dmem_start;
+  const uint8_t *dmem_bss_end;
   /**
-   * End of OTBN data memory.
+   * Offset of the BSS section within DMEM.
    */
-  const uint8_t *dmem_end;
+  const uint8_t *dmem_bss_offset;
```

Comment:
This should be a uint32_t, I think: it's an absolute value giving an OTBN address.

Reply:
Oh good point. But can't you just do the cast to a `uint32_t` in the body of OTBN_APP_T_INIT?

Reply:
Good point, that works! I do have to remove the `const` qualifier from this line for it to compile, though, otherwise the `ctx->app = app;` line from `otbn_load_app` fails with `cannot assign to non-static data member 'app' with const-qualified data member 'dmem_bss_offset'`. It seems to me like that change is probably worth it; just noting it here in case there are objections/new ideas for how to get around that.

Reply:
Yep, a `const uint32_t` is a 32-bit integer that you can't assign to. A `const uint8_t*` is a slangy shorthand for `uint8_t const*` (if I've got that right!) which means a pointer you can't write **through**.